### ISSUE TITLE
Release GGL-Net/MSDA-Net artifacts (models, dataset) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [YuChuang1205/MSDA-Net#4]
Hi @YuChuang1205 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv, specifically your paper "Gradient-Guided Learning Network for Infrared Small Target Detection" (https://huggingface.co/papers/2512.09497). If you are one of the authors, you can submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability at https://huggingface.co/papers/submit. The paper page lets people discuss your paper and find related artifacts (models, datasets, demos). You can also claim the paper, which will show up on your public profile at HF, and add GitHub/project page URLs.

I noticed that your code, including for GGL-Net and MSDA-Net, is available on GitHub (`https://github.com/YuChuang1205/MSDA-Net`), and you've also released several pre-trained `MSDA-Net` models and a processed dataset for infrared small target detection via Baidu Pan. It would be great to make these checkpoints and the processed dataset available on the 🤗 Hub, to improve their discoverability and visibility. We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the processed dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗
